### ISSUE TITLE
ReadableStream cancel is not called when client disconnects

### ISSUE THREAD

**Issue by luccas40** [oven-sh/bun#6758]
### What version of Bun is running?

1.0.7

### What platform is your computer?

Debian Bullseye x64

### What steps can reproduce the bug?

Running Bun with Svelte, in a API GET function:
```
    let keepAlive = true;
    const stream = new ReadableStream({
        start: async controller =>{
            while(keepAlive){           
                await new Promise(r => setTimeout(r, 1000));
            }
        },
        cancel: () =>{
            console.log("cancelled");
            keepAlive = false;
        }
    });
    return new Response(stream, {
        headers: {
            "Cache-Control": "no-store",
            "Content-Type": "text/event-stream",
            "Connection": "keep-alive",
        }
    });
```

### What is the expected behavior?

Everytime the page is closed or the connection is closed by the client we should see a log on the server "cancelled".

### What do you see instead?

Nothing is logged and the loop keeps running even if the page is fully closed.

### Additional information

It works just fine with Node

**Comment by giesf** [comment]
Any news on this? :/

**Comment by ksmithut** [comment]
@giesf I'm not exactly sure how it's supposed to work, but I got something working going off of the request abort signal rather than relying on the response cancel callback:

```js
const server = Bun.serve({
  async fetch(req) {
    const stream = new ReadableStream({
      async start(controller) {
        const encoder = new TextEncoder()
        let count = 0
        const interval = setInterval(() => {
          controller.enqueue(encoder.encode(`${count++}\n`))
        }, 1000)
        let closed = false
        function close() {
          if (closed) return
          console.log('aborted')
          closed = true
          clearInterval(interval)
          req.signal.removeEventListener('abort', close)
          controller.close()
        }
        req.signal.addEventListener('abort', close)
        if (req.signal.aborted) return close()
      }
    })
    return new Response(stream)
  }
})
```

Now running `curl localhost:3000` will get a stream of numbers, and when you `ctrl` + `c` you get the `aborted` log and the request artifacts get cleaned up.

Is that what you're looking for?

**Comment by TheArmagan** [comment]
Any updates?

**Comment by pekeler** [comment]
We're using SSE for the "realtime" aspect of our SPA. Once these ReadableStreams are used up, the app freezes (because server event listeners don't get unregistered). So this isn't just a blocker for going to prod, but it's making development super annoying. Can we please have an ETA for a bug fix?

**Comment by Jarred-Sumner** [comment]
This will be fixed in Bun v1.1.27

**Comment by KentWMartinWestMonroe** [comment]
I'm using bun 1.1.43. I used the following (got the idea from [https://github.com/oven-sh/bun/issues/14390#issuecomment-2660028992](https://github.com/oven-sh/bun/issues/14390#issuecomment-2660028992)) to get it to work

```
const registerStreamingRoutes = (app: Express) => {
  app.post(
    "/streaming",
    multiStrategyMiddleware,
    async (req, res: Response) => {
      (async () => {
           .
           .
           .
        const controller = new AbortController();
        let closed = false;
        function close() {
          if (closed) return;
          closed = true;
          req.removeListener("close", close);
          controller.abort("Aborting...");
        }
        req.addListener("close", close);

        await getStreamingResponse(..., controller, ...);
      })();
    }
  );
```
